### PR TITLE
Initial HDFView 3.2.0 publication

### PR DESCRIPTION
**Pull request by t-chaik**
HDFView is a visual tool for browsing and editing HDF4 and HDF5 files.

https://confluence.hdfgroup.org/display/HDFVIEW/HDFView

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://forum.hdfgroup.org/t/make-hdfview-available-on-flathub-org/9476**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. **Link: 
https://github.com/HDFGroup/hdfview/pull/80**

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.hdfgroup.HDFView.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+app-id: org.hdfgroup.HDFView
+branch: stable
```
Comment (hfiguiere) [inline org.hdfgroup.HDFView.yml:2]:
```suggestion
```

File: org.hdfgroup.HDFView.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  env:
+    JAVA_HOME: /usr/lib/sdk/openjdk/jvm/openjdk-17
+    ANT_HOME: /usr/lib/sdk/openjdk/ant
+finish-args:
+  - --env=PATH=/app/bin:/app/jre/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=x11
```
Comment (hfiguiere) [inline org.hdfgroup.HDFView.yml:18]:
```suggestion
  - --socket=wayland
  - --socket=fallback-x11
```

assuming it supports wayland.

File: org.hdfgroup.HDFView.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk/install.sh
+  - name: hdf4
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.hdfgroup.HDFView.yml:25]:
```suggestion
    buildsystem: cmake-ninja
```

File: org.hdfgroup.HDFView.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        url: https://support.hdfgroup.org/ftp/HDF/releases/HDF4.2.15/src/hdf-4.2.15.tar.bz2
+        sha256: bde035ef5a1cd5fdbd0a7f1fa5c17e98bbd599300189ac4d234f16e9bb7bcb12
+    cleanup:
+      - /bin
+      - /include
+      - /share/cmake
+      - /share/hdf4_java_doc
+      - /share/USING_HDF4_CMake.txt
+  - name: hdf5
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.hdfgroup.HDFView.yml:50]:
```suggestion
    buildsystem: cmake-ninja
```

**Comment by hfiguiere** [comment]
bot, build org.hdfgroup.HDFView

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by t-chaik** [comment]
Blind attempt at fixing the HDF4 `aarch64` build using [Fedora patch set](https://src.fedoraproject.org/rpms/hdf/tree/rawhide). Not sure how to credit their work here though...

File: org.hdfgroup.HDFView.yml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      - type: patch
+        path: hdf-ppc.patch
+      - type: patch
+        path: hdf-4.2.4-sparc.patch
+      - type: patch
+        path: hdf-s390.patch
+      - type: patch
+        path: hdf-arm.patch
+      - type: patch
+        path: hdf-aarch64.patch
```
Comment (hfiguiere) [inline org.hdfgroup.HDFView.yml:50]:
```suggestion
      - type: patch
        paths
          - hdf-ppc.patch
          - hdf-4.2.4-sparc.patch
          - hdf-s390.patch
          - hdf-arm.patch
          - hdf-aarch64.patch
```

I'll assume you need all these unsupported achiecture patches to get the aarch64 patch.

(if only they just used stdint)

Comment (t-chaik) [reply org.hdfgroup.HDFView.yml:50]: > I'll assume you need all these unsupported achiecture patches to get the aarch64 patch.

Indeed...

**Comment by hfiguiere** [comment]
FYI, you can also:

bot, build org.hdfgroup.HDFView

**Pull request closed without merging**